File: Domain/Entities/Persons/Boatbuilder.cs
using System;

namespace Domain.Entities.Persons
{
    public class Boatbuilder
    {
        public Guid BoatbuilderId { get; set; }
        public Guid? AssignmentId { get; set; }
        public Shipyard Shipyard { get; set; }
        public string Name { get; set; }
        public int Skill { get; set; }
        public int Resources { get; set; }
        public int Loyalty { get; set; }
        public int Age { get; set; }
    }
}

File: reactui/src/common/models/boatbuilder.ts
import { IShipyard } from "./shipyard";

export interface IBoatbuilder {
    boatbuilderId: string;
    assignedToId: string;
    shipyard: IShipyard;
    name: string;
    resources: number;
    skill: number;
    loyalty: number;
    age: number;
}
